cohortextractor/cohortextractor.py

```diff
@@ -315,7 +315,7 @@ def _generate_measures(
             # already exist we can avoid loading the patient data entirely
             if patient_df is None:
                 patient_df = _load_dataframe_for_measures(filepath, measures)
-            measure_df = measure.calculate(patient_df)
+            measure_df = measure.calculate(patient_df, _report)
```

Comment:
I didn't think of that. It's certainly more "functional", which I like. Certainly simpler to test, maybe a little more clumsy inside `Measure` since you have to wrangle the list of messages (although hardly as it is with only one function actually reporting anything).